packages/bff/src/services/clientService.ts

```diff
@@ -0,0 +1,358 @@
+        params: { clientId, keyId },
+        headers,
+      });
+    },
+
+    async removeUser(
+      clientId: string,
+      userId: string,
+      { logger, headers }: WithLogger<BffAppContext>
+    ): Promise<void> {
+      logger.info(`Removing user ${userId} from client ${clientId}`);
+
+      return authorizationProcessClient.client.removeUser(undefined, {
+        params: { clientId, userId },
+        headers,
+      });
+    },
+
+    async addUserToClient(
+      userId: string,
+      clientId: string,
+      { logger, headers }: WithLogger<BffAppContext>
+    ): Promise<void> {
+      logger.info(`Add user ${userId} to client ${clientId}`);
+
+      await authorizationProcessClient.client.addUser(undefined, {
+        params: { clientId, userId },
+        headers,
+      });
+    },
+
+    async createKeys(
+      userId: string,
+      clientId: string,
+      keySeed: bffApi.KeysSeed,
+      { logger, headers }: WithLogger<BffAppContext>
+    ): Promise<void> {
+      logger.info(`Create keys for client ${clientId}`);
+
+      const body: authorizationApi.KeysSeed = keySeed.map((seed) => ({
+        userId,
+        key: seed.key,
+        use: seed.use,
+        alg: seed.alg,
+        name: seed.name,
+        createdAt: new Date().toISOString(),
+      }));
+
+      await authorizationProcessClient.client.createKeys(body, {
+        params: { clientId },
+        headers,
+      });
+    },
+
+    async getClientKeys(
+      clientId: string,
+      userIds: string[],
+      { logger, headers }: WithLogger<BffAppContext>
+    ): Promise<bffApi.PublicKey[]> {
+      logger.info(`Retrieve keys of client ${clientId}`);
+
+      const { keys } = await authorizationProcessClient.client.getClientKeys({
+        params: { clientId },
+        queries: { userIds: userIds.join(",") },
+        headers,
+      });
+
+      return Promise.all(
+        keys.map((k) => decorateKey(selfcareUsersClient, k, clientId))
+      );
+    },
+
+    async addClientPurpose(
+      clientId: string,
+      purpose: bffApi.PurposeAdditionDetailsSeed,
+      { logger, headers }: WithLogger<BffAppContext>
+    ): Promise<void> {
+      logger.info(`Adding purpose ${purpose.purposeId} to client ${clientId}`);
+
+      await authorizationProcessClient.client.addClientPurpose(purpose, {
+        params: { clientId },
+        headers,
+      });
+    },
+
+    async getClientUsers(
+      clientId: string,
+      selfcareId: string,
+      { logger, headers }: WithLogger<BffAppContext>
+    ): Promise<bffApi.CompactUser[]> {
+      logger.info(`Retrieving users for client ${clientId}`);
+
+      const clientUsers =
+        await authorizationProcessClient.client.getClientUsers({
+          params: { clientId },
+          headers,
+        });
+
+      const users = clientUsers.map(async (id) =>
+        toBffApiCompactUser(
+          await getSelfcareUserById(selfcareUsersClient, id, selfcareId),
```

Comment:
Also here in Scala it calls the same `getSelfcareUserById` that does not throw, see other comment about this in `decorateKey`. So I think also in this case you should do the same and no need to handle `userNotFound` in the error mapper